### PR TITLE
mingw-builds: added 13.2.0

### PR DESCRIPTION
**Pull request by elvisdukaj**
Specify library name and version:  **mingw-builds/13.2.0**

I added MinGW 13.2.0 version. Note that this version adds a new runtime UCRT runtime options that I set as default as suggested in the [MinGW documentation](https://www.msys2.org/docs/environments/)

> UCRT (Universal C Runtime) is a newer version which is also used by Microsoft Visual Studio by default. It should work and behave as if the code was compiled with MSVC.
>
> - Better compatibility with MSVC, both at build time and at run time.
> - It is included as part of the operating system in Windows 10 or later, and Windows Server 2016 or later. For older Windows versions, you have to provide it explicitly or depend on the user having it installed. More information about that can be found in [UCRT deployment](https://learn.microsoft.com/en-us/cpp/windows/universal-crt-deployment).

On the other hand:
> MSVCRT (Microsoft Visual C++ Runtime) is available by default on all Microsoft Windows versions, but due to backwards compatibility issues is stuck in the past, not C99 compatible and is missing some features.

> - It isn't C99 compatible, for example the printf() function family, but...
> - mingw-w64 provides replacement functions to make things C99 compatible in many cases
> - It doesn't support the UTF-8 locale
> - Binaries linked with MSVCRT should not be mixed with UCRT ones because the internal structures and data types are different. (More strictly, object files or static libraries built for different targets shouldn't be mixed. DLLs built for different CRTs can be mixed as long as they don't share CRT objects, e.g. FILE*, across DLL boundaries.) Same rule is applied for MSVC compiled binaries because MSVC uses UCRT by default (if not changed).
> - Works out of the box on every version of Microsoft Windows.

Maybe a better option could use by default UCRT with Windows 10+ and MSVCRT otherwise, is any option to check the version of Windows?
---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by elvisdukaj** [comment]
I don't know why the pipeline is failing 🤔any chance to read the logs?

File: recipes/mingw-builds/all/conanfile.py [hunk 1]
```diff
@@ -14,8 +15,16 @@ class MingwConan(ConanFile):
     license = "ZPL-2.1", "MIT", "GPL-2.0-or-later"
     topics = ("gcc", "gnu", "unix", "mingw32", "binutils")
     settings = "os", "arch"
-    options = {"threads": ["posix", "win32"], "exception": ["seh", "sjlj"]}
-    default_options = {"threads": "posix", "exception": "seh"}
+    options = {
+        "threads": ["posix", "win32", "mcf"],
```
[the remark this answers is not in the document]
Comment (elvisdukaj) [reply recipes/mingw-builds/all/conanfile.py:19]: Good catch!

File: recipes/mingw-builds/all/conanfile.py [hunk 2]
```diff
@@ -14,8 +15,16 @@ class MingwConan(ConanFile):
...
+        "exception": ["seh", "sjlj"],
+        "runtime": ["msvcrt", "ucrt"]
```
[the remark this answers is not in the document]
Comment (elvisdukaj) [reply recipes/mingw-builds/all/conanfile.py:21]: So with 12.0v10r2 it was introduced the `ucrt` runtime. Before that `msvcrt` was the only choice.

How do you handle situation where the default value of an option changes in base of the version of the package?

[the remark this answers is not in the document]
Comment (elvisdukaj) [reply recipes/mingw-builds/all/conanfile.py:21]: > Maybe a null value should be introduced for older versions, so it doesn't affect the package_id?

Good idea :)

[the remark this answers is not in the document]
Comment (elvisdukaj) [reply recipes/mingw-builds/all/conanfile.py:21]: I am not sure if it's the right approach but on the `validate()` method I remove the runtime options for the versions <= 12.2.0. This way the `package_id` shouldn't change. How can I make a quick check to validate this?

[the remark this answers is not in the document]
Comment (elvisdukaj) [reply recipes/mingw-builds/all/conanfile.py:21]: Unfortunately the `package_id` are different also removing the `runtime`:

Using `conan list mingw-builds/12.2.0:* -r conancenter`:
```
conancenter
  mingw-builds
    mingw-builds/12.2.0
      revisions
        6d4f9fa696c4097c55762e2f481a2355 (2023-07-06 11:14:43 UTC)
          packages
            f4943bfd33cc2961ebbb1d81bf8b090153522e1f
              info
                settings
                  arch: x86_64
                  os: Windows
                options
                  exception: seh
                  threads: posix
```

While my local package is:
```
local
  mingw-builds
    mingw-builds/12.2.0
      revisions
        e24bc714576c31c8f083b580dae5257b (2024-04-17 14:31:06 UTC)
          packages
            f82b765f5ccc390bb8d409c50689a5c5ab3eb642
              info
                settings
                  arch: x86_64
                  os: Windows
                options
                  exception: seh
                  runtime: ucrt
                  threads: posix
```

Any idea on how to fix this?

[the remark this answers is not in the document]
Comment (elvisdukaj) [reply recipes/mingw-builds/all/conanfile.py:21]: Ok it was as simple as doing this during the `configure()` method. Now the package_id are identical for the previous releases:

```
$ conan list mingw-builds/12.2.0:* -r local
local
  mingw-builds
    mingw-builds/12.2.0
      revisions
        bf2318e4f907e60965dfcd7c85575f43 (2024-04-17 15:11:11 UTC)
          packages
            f4943bfd33cc2961ebbb1d81bf8b090153522e1f
              info
                settings
                  arch: x86_64
                  os: Windows
                options
                  exception: seh
                  threads: posix
                  
$ conan list mingw-builds/12.2.0:* -r conancenter
conancenter
  mingw-builds
    mingw-builds/12.2.0
      revisions
        6d4f9fa696c4097c55762e2f481a2355 (2023-07-06 11:14:43 UTC)
          packages
            f4943bfd33cc2961ebbb1d81bf8b090153522e1f
              info
                settings
                  arch: x86_64
                  os: Windows
                options
                  exception: seh
                  threads: posix

$ conan list mingw-builds/13.2.0:* -r local      
local
  mingw-builds
    mingw-builds/13.2.0
      revisions
        e24bc714576c31c8f083b580dae5257b (2024-04-17 15:11:13 UTC)
          packages
            f82b765f5ccc390bb8d409c50689a5c5ab3eb642
              info
                settings
                  arch: x86_64
                  os: Windows
                options
                  exception: seh
                  runtime: ucrt
                  threads: posix                  
```

**Comment by elvisdukaj** [comment]
Note that StrawberryPerl has a similar discussion: https://github.com/StrawberryPerl/Perl-Dist-Strawberry/issues/104

**Comment by elvisdukaj** [comment]
Also the threading model could be changed to use mcf instead of posix, it seemed much more performant and suitable for GCC.

**Comment by elvisdukaj** [comment]
Any opinion about this PR?

**Comment by jcar87** [comment]
Hi @elvisdukaj  - we are still in the process of reviewing this.

Please avoid updating the branch without making changes to the recipe - it triggers the CI process again for the new commit, using compute resources unnecessarily.

**Review by ericLemanissier — APPROVED** [review]

**Review by memsharded — APPROVED** [review]

**Pull request merged**